android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt

```diff
@@ -303,255 +321,295 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         val trackList = Arguments.toList(data)
         val queue = musicService.tracks
 
-        if (trackList != null) {
-            musicService.getCurrentTrackIndex {
+        scope.launch {
+            if (trackList != null) {
+                val currentIndex = musicService.getCurrentTrackIndex()
+
                 for (track in trackList) {
                     val index = if (track is Int) track else track.toString().toInt()
 
                     // We do not allow removal of the current item
-                    if (index == it) {
+                    if (index == currentIndex) {
                         Timber.e("This track is currently playing, so it can't be removed")
-                        return@getCurrentTrackIndex
+                        continue
```

Comment:
Not sure if this should be a `continue` or a `break`

Reply:
Based on iOS a continue is correct:

https://github.com/doublesymmetry/react-native-track-player/blob/main/ios/RNTrackPlayer/RNTrackPlayer.swift#L366